### PR TITLE
FOUNDATIONS: Modify Company

### LINKED ISSUE #197: FOUNDATIONS: Modify `Company`
NOTE #196 needs to be completed and merged first

### PR DESCRIPTION
**Pull request by sulton-max**
Closes #197

### CODE REVIEW

File: CashOverflow.Tests.Unit/Services/Foundations/Companies/CompanyServiceTests.Exceptions.Modify.cs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            SqlException sqlException = CreateSqlException();
+
+            var failedCompanyStorageException = 
+                new FailedCompanyStorageException(sqlException);
+            
+            var expectedCompanyDependencyException = 
+                new CompanyDependencyException(failedCompanyStorageException);
+
+            this.storageBrokerMock.Setup(broker => 
+                broker.SelectCompanyByIdAsync(companyId))
```
[the remark this answers is not in the document]
Comment (sulton-max) [reply CashOverflow.Tests.Unit/Services/Foundations/Companies/CompanyServiceTests.Exceptions.Modify.cs:37]: Company da UpdatedDate yo'q bo'lganligi uchun DateTime broker kerakmas deb o'ylagandim, chunki CreatedDate tekshirilmaydi Modifyda to'g'rimi ?

File: CashOverflow/Models/Companies/Exceptions/CompanyServiceException.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+using System;
+using Xeptions;
+
+namespace CashOverflow.Models.Companies.Exceptions
+{
+    public class CompanyServiceException : Xeption
+    {
+        public CompanyServiceException(Exception innerException) 
+            : base(message: "Company service error occurred, please contact support", innerException)
```
Comment (ElbekDeveloper) [inline CashOverflow/Models/Companies/Exceptions/CompanyServiceException.cs:14]: be consisted. Remove `please` from here

File: CashOverflow/Models/Companies/Exceptions/FailedCompanyServiceException.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+using System;
+using Xeptions;
+
+namespace CashOverflow.Models.Companies.Exceptions
+{
+    public class FailedCompanyServiceException : Xeption
+    {
+        public FailedCompanyServiceException(Exception innerException) 
+            : base(message: "Failed company service error occurred, please contact support.", innerException)
```
Comment (ElbekDeveloper) [inline CashOverflow/Models/Companies/Exceptions/FailedCompanyServiceException.cs:14]: remove `please`

File: CashOverflow/Models/Companies/Exceptions/LockedCompanyException.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+using System;
+using Xeptions;
+
+namespace CashOverflow.Models.Companies.Exceptions
+{
+    public class LockedCompanyException : Xeption
+    {
+        public LockedCompanyException(Exception innerException) 
+            : base(message: "Company is locked, please try again", innerException)
```
Comment (ElbekDeveloper) [inline CashOverflow/Models/Companies/Exceptions/LockedCompanyException.cs:14]: remove please and add `later`. So, if the object is locked, they should give some time

Comment (ElbekDeveloper) [reply CashOverflow/Models/Companies/Exceptions/LockedCompanyException.cs:14]: Also add dot (.)

**Pull request merged**